BrainCloudClient/Assets/BrainCloud/Client/BrainCloud/Internal/FileUploader.cs

```diff
@@ -169,18 +163,18 @@ public void Start()
             };
 
             var requestContent = new MultipartFormDataContent();
-
-            ProgressStream fileStream = new ProgressStream(new FileStream(_localPath, FileMode.Open, FileAccess.Read, FileShare.Read));
+            
+            ProgressStream fileStream = new ProgressStream(new MemoryStream(_client.FileService.FileStorage[_localPath]));
             fileStream.BytesRead += BytesReadCallback;
 
             requestContent.Add(new StringContent(_sessionId), "sessionId");
             if (_peerCode != "") requestContent.Add(new StringContent(_peerCode), "peerCode");
             requestContent.Add(new StringContent(UploadId), "uploadId");
             requestContent.Add(new StringContent(TotalBytesToTransfer.ToString()), "fileSize");
             requestContent.Add(new StreamContent(fileStream), "uploadFile", _fileName);
-
+            
             requestMessage.Content = requestContent;
-
+            _client.FileService.FileStorage.Remove(_localPath);
```

Comment:
Can you move this line after fetching it at line 167? This way if something goes wrong, at least we removed it already.

```
byte[] fileData = _client.FileService.FileStorage[_localPath];
if (fileData == null) throw error!
_client.FileService.FileStorage.Remove(_localPath);
ProgressStream fileStream = new ProgressStream(new MemoryStream(fileData ));
...
```

Reply:
^ updated comment